front/app/containers/Admin/reporting/components/utils.test.ts

```diff
@@ -0,0 +1,72 @@
+import { IAppConfigurationData } from 'services/appConfiguration';
+import { shouldShowReportBuilderTab } from './utils';
+
+describe('shouldShowReportBuilderTab', () => {
+  it('should return true if report_builder feature flag is allowed and enabled', () => {
+    const appConfig = {
+      attributes: {
+        settings: {
+          report_builder: {
+            allowed: true,
+            enabled: true,
+          },
+        },
+      },
+    };
+
+    expect(shouldShowReportBuilderTab(appConfig as IAppConfigurationData)).toBe(
+      true
+    );
+  });
+
+  it('should return true if report_builder feature flag is not allowed and enabled', () => {
```

Comment:
```suggestion
  it('should return true if report_builder feature flag is not allowed but is enabled', () => {
```
Suggestion to change the wording because now I understood it like 'is not allowed and not enabled'.

In reality we should never get in this situation. If allowed is false, enabled should be false too. You can add that as a comment.